packages/backend-storage/src/storage_access_policy_factory.ts

```diff
@@ -63,9 +63,15 @@ export class StorageAccessPolicyFactory {
       });
     }
 
-    return new Policy(this.stack, `${this.namePrefix}${this.policyCount++}`, {
-      statements,
-    });
+    return new Policy(
+      this.stack,
+      `${this.bucket.bucketName.slice(0, 6)}${Math.floor(
```

Comment:
same here, this needs consistent hashing or global counter.

Reply:
done 2f932ec1fd42f741a56d36e2c04096d81dae0711

Reply:
- [ ] need to revisit

Reply:
Done. Using global counter ec113d0077